shared/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.test.js

```diff
@@ -285,6 +285,40 @@ describe('fileAndServeCourtIssuedDocumentInteractor', () => {
     ).toHaveBeenCalled();
   });
 
+  it('should set the leadDocketNumber for work items', async () => {
+    const leadDocketNumber = MOCK_CASE.docketNumber;
+    const caseRecordWithLeadDocketNumber = {
+      ...MOCK_CASE,
+      docketEntries: [mockDocketEntryWithWorkItem],
+      leadDocketNumber,
+    };
+
+    applicationContext
+      .getPersistenceGateway()
+      .getCaseByDocketNumber.mockReturnValueOnce(caseRecordWithLeadDocketNumber)
+      .mockReturnValueOnce(caseRecordWithLeadDocketNumber);
```

Comment:
❓ Why do we use mockReturnValueOnce two times with the same return but no default?

Reply:
My understanding is mockReturnValueOnce is needed to be called twice in a row because in the interactor itself getCaseByDocketNumber is invoked once for subjectCase and then invoked a second time in the try catch block.  Since it gets invoked a second time, when the test passes in caseRecordWithLeadDocketNumber, the second invocation causes the leadDocketNumber to become undefined and therefore fails.

Reply:
After talking through this, I believe the reason we are doing this is because the default mock would persist into the later tests and have unintended consequences.